iob_soc.py

```diff
@@ -33,325 +34,325 @@
+                    "bits": [],
+                },
             ),
-            iob_block_group(
-                name="mem",
-                description="Memory module",
-                blocks=[merge.instance("iob_merge_0"), merge.instance("iob_merge_1")],
+            (
+                {
+                    "corename": "UART0",
+                    "if_name": "rs232",
+                    "port": "cts",
+                    "bits": [],
+                },
+                {
+                    "corename": "external",
+                    "if_name": "uart",
+                    "port": "uart_cts_i",
+                    "bits": [],
+                },
             ),
-            iob_block_group(
-                name="peripheral",
-                description="Peripheral module",
-                blocks=self.peripherals,
+            (
+                {
+                    "corename": "UART0",
+                    "if_name": "rs232",
+                    "port": "rts",
+                    "bits": [],
+                },
+                {
+                    "corename": "external",
+                    "if_name": "uart",
+                    "port": "uart_rts_o",
+                    "bits": [],
+                },
             ),
         ]
+    )
+    # Number of external memory connections (will be filled automatically)
+    num_extmem_connections = -1
+    # This is a standard iob_module attribute, but needs to be defined after 'peripherals' because it depends on it
+    block_groups = [
+        iob_block_group(
+            name="cpu", description="CPU module", blocks=[cpu.instance("cpu_0")]
+        ),
+        iob_block_group(
+            name="bus_split",
+            description="Split modules for buses",
+            blocks=[
+                split.instance("ibus_split_0"),
+                split.instance("dbus_split_0"),
+                split.instance("int_dbus_split_0"),
+                split.instance("pbus_split_0"),
+            ],
+        ),
+        iob_block_group(
+            name="mem",
+            description="Memory module",
+            blocks=[merge.instance("iob_merge_0"), merge.instance("iob_merge_1")],
+        ),
+        iob_block_group(
+            name="peripheral",
+            description="Peripheral module",
+            blocks=self.peripherals,
+        ),
+    ]
 
-    def _setup(self, *args, is_top=True, **kwargs):
+    def __post_init__(self, *args, is_top=True, **kwargs):
         self.is_top_module = is_top
         self.set_default_build_dir()
+        # TODO: Somehow we have to run pre_setup before the __init__
         # Pre-setup specialized IOb-SoC functions
         pre_setup_iob_soc(self)
-        # Call the superclass _setup
-        super()._setup(*args, is_top=is_top, **kwargs)
+        # Call the superclass _setup # TODO: DELETE THIS. This does not exist anymore. Only leaving here for reference.
+        # super()._setup(*args, is_top=is_top, **kwargs)
```

Comment:
I'm not currently fully aware of how the dataclasses interact with subclasses, but I assume that the `__post_init__` method will override the `__post_init__()` from the superclass.
Therefore, I think this line will probably become:
```suggestion
        super().__post_setup__(*args, is_top=is_top, **kwargs)
```

Reply:
Do you mean __post_init__?

Reply:
pls try in a simple example , ask chatgpt...